libvncclient/corre.c

```diff
@@ -46,7 +46,10 @@ HandleCoRREBPP (rfbClient* client, int rx, int ry, int rw, int rh)
     if (!ReadFromRFBServer(client, (char *)&pix, sizeof(pix)))
 	return FALSE;
 
-    FillRectangle(client, rx, ry, rw, rh, pix);
+    if(client->GotFillRect != NULL)
```

Comment:
Just a thought: You could get rid of those ifs by connecting the original functions to the hooks in vncviewer.c:rfbGetClient()